pkg/fleet/installer/service/apm_sockets.go

```diff
@@ -0,0 +1,161 @@
+		apmSocket = apmOldSocket
+	}
+	if cfg.DogstatsdSocket == statsdOldSocket {
+		statsdSocket = statsdOldSocket
+	}
+	return apmSocket, statsdSocket
+}
+
+// configureSocketsEnv configures the sockets for the agent & injector
+func configureSocketsEnv() error {
+	envFile := newFileMutator("/etc/environment", setSocketEnvs, verifySocketEnvs, verifySocketEnvs)
+	defer envFile.cleanup()
+	rollback, err := envFile.mutate()
+	if err != nil {
+		rollbackErr := rollback()
+		if rollbackErr != nil {
+			log.Warnf("Failed to rollback environment file: %v", rollbackErr)
+		}
+		return fmt.Errorf("error configuring sockets: %w", err)
+	}
+	return nil
+}
+
+// setSocketEnvs sets the socket environment variables
+func setSocketEnvs(envFile []byte) ([]byte, error) {
+	apmSocket, statsdSocket := getSocketsPath(agentConfigPath, apmInjectOldPath)
+	envs := map[string]string{
+		"DD_APM_RECEIVER_SOCKET": apmSocket,
+		"DD_DOGSTATSD_SOCKET":    statsdSocket,
+		"DD_USE_DOGSTATSD":       "true",
+	}
+	return addEnvsIfNotSet(envs, envFile)
+}
+
+// addEnvsIfNotSet adds environment variables to the environment file if they are not already set
+func addEnvsIfNotSet(envs map[string]string, envFile []byte) ([]byte, error) {
+	var buffer bytes.Buffer
+	buffer.Write(envFile)
+
+	if len(envFile) > 0 && envFile[len(envFile)-1] != '\n' {
+		buffer.WriteByte('\n')
+	}
+	for key, value := range envs {
+		// There's a slight gap where there exists an env var in the form of {prefix}{key} but not {key},
+		// but as our env vars all start with DD_ prefixing them won't cause any issue.
+		if !bytes.Contains(envFile, []byte(key+"=")) {
+			buffer.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+		}
+	}
+	return buffer.Bytes(), nil
+}
+
+// verifySocketEnvs verifies that socket environment variables can be sourced
+func verifySocketEnvs(path string) error {
+	cmd := exec.Command("bash", "-c", fmt.Sprintf("source %s && echo $DD_APM_RECEIVER_SOCKET $DD_DOGSTATSD_SOCKET $DD_USE_DOGSTATSD", path))
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error verifying socket environment variables: %w", err)
+	}
+	if string(output) != fmt.Sprintf("%s %s true\n", apmInstallerSocket, statsdInstallerSocket) {
+		return fmt.Errorf("socket environment variables not set correctly: %s", output)
+	}
+	return nil
+}
+
+// addSystemDEnvOverrides adds /etc/environment variables to the defined systemd units
+// The unit should contain the .service suffix (e.g. datadog-agent-exp.service)
+//
+// Reloading systemd & restarting the unit has to be done separately by the caller
+func addSystemDEnvOverrides(unit string) error {
+	// Verify that the unit is a DD one
+	if !strings.HasPrefix(unit, "datadog-") {
+		return fmt.Errorf("unit %s is not a Datadog unit", unit)
+	}
+	if !strings.HasSuffix(unit, ".service") {
+		return fmt.Errorf("unit %s is not a service unit", unit)
+	}
+
+	content := []byte("[Service]\nEnvironmentFile=/etc/environment\n")
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏TBH I wonder if we should use `/etc/datadog-agent/environment` instead, given that it's already loaded in the deb core-agent & in all installer agent units. It just won't be sourced for interactive sessions, which should be fine anyway.